SU2_CFD/src/solver_structure.cpp

```diff
@@ -4825,6 +4825,365 @@ void CSolver::Read_InletFile_ASCII(CGeometry *geometry, CConfig *config, string
   
 }
 
+void CSolver::PrintNewInletData (passivedouble *Inlet_InterpolatedData, unsigned long nVertex,string Marker_Tag)
+{
+
+ofstream myfile;
+myfile.precision(16);
+myfile.open("Interpolated_Data_"+Marker_Tag+".dat",ios_base::out);
+
+
+if (myfile.is_open())
+{  
+            for (unsigned long iVertex = 0; iVertex < nVertex; iVertex++) {
+
+                  for  (unsigned long iVar=0; iVar < (maxCol_InletFile+nDim); iVar++)
+                    {
+                      myfile<<Inlet_InterpolatedData[iVertex*(maxCol_InletFile+nDim)+iVar]<<"\t";
+                    }
+                  myfile<<endl;
+                }
+        myfile.close();
+}
+  else
+    cout<<"file cannot be opened"<<endl;
+}
+
+
+void CSolver::LoadSpanwiseInletProfile(CGeometry **geometry,
+                                CSolver ***solver,
+                                CConfig *config,
+                                int val_iter,
+                                unsigned short val_kind_solver,
+                                unsigned short val_kind_marker)
+{
+  /*-- First, set the solver and marker kind for the particular problem at
+   hand. Note that, in the future, these routines can be used for any solver
+   and potentially any marker type (beyond inlets). ---*/
+
+  unsigned short KIND_SOLVER = val_kind_solver;
+  unsigned short KIND_MARKER = val_kind_marker;
+
+  /*--- Local variables ---*/
+
+  unsigned short iDim, iVar, iMesh, iMarker, jMarker;
+  unsigned long iPoint, iVertex, iChildren, Point_Fine, iRow, nVertex;
+  passivedouble ai,bi,ci,di,delta,dxi; //akima interpolation variables
+  passivedouble Theta, Parameter1, Parameter2, unit_r, unit_Theta, unit_m, unit_z, Alpha, Phi;
+  su2double Area_Children, Area_Parent, *Coord, dist, slope, Interp_Radius;
```

Comment:
When moving the interpolation to a class make all parameters su2double as we might want to differentiate through it (it can also be a template class).

Reply:
Ok, will do.